arch/arc/core/dsp/swap_dsp_macros.h

```diff
@@ -0,0 +1,262 @@
+	lr r13, [_ARC_V2_AGU_MOD2]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod2_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD3]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod3_OFFSET]
+#ifdef CONFIG_ARC_AGU_MEDIUM
+	lr r13, [_ARC_V2_AGU_AP4]
+	st r13, [sp, ___callee_saved_stack_t_agu_ap4_OFFSET]
+	lr r13, [_ARC_V2_AGU_AP5]
+	st r13, [sp, ___callee_saved_stack_t_agu_ap5_OFFSET]
+	lr r13, [_ARC_V2_AGU_AP6]
+	st r13, [sp, ___callee_saved_stack_t_agu_ap6_OFFSET]
+	lr r13, [_ARC_V2_AGU_AP7]
+	st r13, [sp, ___callee_saved_stack_t_agu_ap7_OFFSET]
+	lr r13, [_ARC_V2_AGU_OS2]
+	st r13, [sp, ___callee_saved_stack_t_agu_os2_OFFSET]
+	lr r13, [_ARC_V2_AGU_OS3]
+	st r13, [sp, ___callee_saved_stack_t_agu_os3_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD4]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod4_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD5]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod5_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD6]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod6_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD7]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod7_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD8]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod8_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD9]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod9_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD10]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod10_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD11]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod11_OFFSET]
+#endif
+#ifdef CONFIG_ARC_AGU_LARGE
+	lr r13, [_ARC_V2_AGU_AP8]
+	st r13, [sp, ___callee_saved_stack_t_agu_ap8_OFFSET]
+	lr r13, [_ARC_V2_AGU_AP9]
+	st r13, [sp, ___callee_saved_stack_t_agu_ap9_OFFSET]
+	lr r13, [_ARC_V2_AGU_AP10]
+	st r13, [sp, ___callee_saved_stack_t_agu_ap10_OFFSET]
+	lr r13, [_ARC_V2_AGU_AP11]
+	st r13, [sp, ___callee_saved_stack_t_agu_ap11_OFFSET]
+	lr r13, [_ARC_V2_AGU_OS4]
+	st r13, [sp, ___callee_saved_stack_t_agu_os4_OFFSET]
+	lr r13, [_ARC_V2_AGU_OS5]
+	st r13, [sp, ___callee_saved_stack_t_agu_os5_OFFSET]
+	lr r13, [_ARC_V2_AGU_OS6]
+	st r13, [sp, ___callee_saved_stack_t_agu_os6_OFFSET]
+	lr r13, [_ARC_V2_AGU_OS7]
+	st r13, [sp, ___callee_saved_stack_t_agu_os7_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD12]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod12_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD13]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod13_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD14]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod14_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD15]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod15_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD16]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod16_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD17]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod17_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD18]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod18_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD19]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod19_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD20]
+	st r13, [sp, ___callee_saved_stack_t_agu_mod20_OFFSET]
+	lr r13, [_ARC_V2_AGU_MOD21]
+	_st32_huge_offset r13, sp, ___callee_saved_stack_t_agu_mod21_OFFSET, r1
```

Comment:
I'm really curious about this - for both st and ld we have same offset size here (s9 value). So how we do 
`ld r13, [sp, ___callee_saved_stack_t_agu_mod21_OFFSET]` 
but can't do 
`st r13, [sp, ___callee_saved_stack_t_agu_mod21_OFFSET]`

?